sklearn/ensemble/_stacking.py

```diff
@@ -156,31 +163,40 @@ def fit(self, X, y, sample_weight=None):
             else:
                 self.named_estimators_[name_est] = 'drop'
 
-        # To train the meta-classifier using the most data as possible, we use
-        # a cross-validation to obtain the output of the stacked estimators.
-
-        # To ensure that the data provided to each estimator are the same, we
-        # need to set the random state of the cv if there is one and we need to
-        # take a copy.
-        cv = check_cv(self.cv, y=y, classifier=is_classifier(self))
-        if hasattr(cv, 'random_state') and cv.random_state is None:
-            cv.random_state = np.random.RandomState()
-
         self.stack_method_ = [
             self._method_name(name, est, meth)
             for name, est, meth in zip(names, all_estimators, stack_method)
         ]
-        fit_params = ({"sample_weight": sample_weight}
-                      if sample_weight is not None
-                      else None)
-        predictions = Parallel(n_jobs=self.n_jobs)(
-            delayed(cross_val_predict)(clone(est), X, y, cv=deepcopy(cv),
-                                       method=meth, n_jobs=self.n_jobs,
-                                       fit_params=fit_params,
-                                       verbose=self.verbose)
-            for est, meth in zip(all_estimators, self.stack_method_)
-            if est != 'drop'
-        )
+
+        # To train the meta-classifier using the most data as possible, we use
+        # a cross-validation to obtain the output of the stacked estimators.
+
+        if self.cv == "prefit":
+            # Generate predictions from prefit models
+            predictions = Parallel(n_jobs=self.n_jobs)(
```

Comment:
For most estimators, the overhead of parallelism would not be worthwhile. But if we keep this, I think we can define `_predict_single_estimator` locally, rather than in `_base.py`

Reply:
I have moved it locally and switched removed the parallel part.